Dev add utilities

ThorsHamster opened the pull request:
Adds first utility function for Bytes.

Manipulation of Bytes will be useful for several packages. Therefore a utility-functionality could be used. This first draft only adds functionality, which will already be used inside of IMU. We may append this in the future will further useful functionalities.

Note: It is not possible to add this branch in codacy. Reason is not known.

ThorsHamster committed:
Initial draft for Byte Class

ThorsHamster committed:
Removal of space

ThorsHamster committed:
Add of byte_initialize_with_number

ThorsHamster committed:
Removal of unnecessary tests

ThorsHamster committed:
Add of SetBit

ThorsHamster committed:
Add of SetBitInByte and ClearBit

ThorsHamster committed:
Add of byte_init_0_set_bit_9

ThorsHamster committed:
Use utilities::Byte Class instead of local methods

elheck commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SquidSquad/Squiddrone/pullRequest?prid=6098018&bid=19963376)

elheck commented on src/utilities/byte.hpp:
```diff
@@ -0,0 +1,26 @@
+  ~Byte() = default;
+
+  Byte(){};
+  Byte(std::uint8_t byte) : byte_(byte){};
```
Codacy found an issue: [Class 'Byte' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/SquidSquad/Squiddrone/pullRequest?prid=6098018)

elheck replied:
Codacy found an issue: [Class 'Byte' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/SquidSquad/Squiddrone/pullRequest?prid=6098018)

ThorsHamster committed:
Declare constructor explicitly

Removal of option for empty init. Maybe add a Set() method later, if necessary.

elheck commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SquidSquad/Squiddrone/pullRequest?prid=6098018&bid=19963376)

ThorsHamster commented:
I reopen this pullrequest, cause codacy has some problems with updates through PRs.
